### ISSUE TITLE
Some error messages of type mismatching get wrong positions

### ISSUE THREAD

**Issue by gdh1995** [microsoft/TypeScript#32179]
**TypeScript Version:**  3.6.0-dev.20190629 ; 3.5.1 ; 3.5.2

**Search Terms:** noImplicitThis error position  assignable parameter ; error report wrong position   assignable parameter

**Code**

`a.ts`:
```ts

interface Foo { bar: boolean; }

var GlobalIns = {
  prop1: 1,
  prop2: 2,
  prop3: 3,
  test () {
    this.accept_foo(this);
  },
  accept_foo (foo: Foo): boolean {
    return !!foo && !!foo.bar;
  }
};
```

**Expected behavior:**

Run `tsc --noImplicitThis a.ts` in shell, and it should report a mismatching between the type of `.accept_foo`'s first parameter and the real argument `this` - on the line of `this.accept_foo(this)`.

**Actual behavior:**

The real error message is:
```
a.ts:4:3 - error TS2345: Argument of type '{ prop1: number; prop2: number; prop3: number; test(): void; accept_foo(foo: Foo): boolean; }' is not assignable to parameter of type 'Foo'.
  Object literal may only specify known properties, and 'prop1' does not exist in type 'Foo'.

4   prop1: 1,
    ~~~~~~~~


Found 1 error.
```

The error message shown in VS Code is also located on the first property of `GlobalIns`, but not the real calling of `.accept_foo`;

**Playground Link:** (None)

**Related Issues:** No

**Comment by RyanCavanaugh** [comment]
This shouldn't be an error at all IMO.

Smaller repro:
```ts
const test = {
    x: 10,
    caller() {
        this.check(this);
    },
    check(self: { x: number }) { }
};
```

**Comment by weswigham** [comment]
@RyanCavanaugh you've got a potentially different issue there - the OP is getting an excess property error on the _containing object_ (lol) - he _should_ have an error - he's passing something without a `bar` to a thing that needs a `bar`.

**Comment by gdh1995** [comment]
Yes.

And a real case is that I ran into such a "confusing" error on 06/29, and the TS file defines a global object with 1600+ lines, the argument mismatching occurred after I modified the argument types of one member function, but forgot to modify all the calling places - after a while, I was totally missing in the reported error. Finally, I had to delete member functions in the file one by one, and watch whether the error would disappear after tens of lines being deleted.

**Comment by fatcerberus** [comment]
> the OP is getting an excess property error on the *containing object*

Yay for tautological error messages!  My guess is this is some weird interaction between excess property checking and contextual typing?
